shell/promptRemove/pod.vue

```diff
@@ -0,0 +1,149 @@
+<script>
+import { Banner } from '@components/Banner';
+import Checkbox from '~/pkg/rancher-components/src/components/Form/Checkbox/Checkbox.vue';
+import { mapGetters, mapState } from 'vuex';
+import { isEmpty } from 'lodash';
+
+export default {
+  name: 'PromptRemovePodDialog',
+
+  components: {
+    Banner,
+    Checkbox
+  },
+
+  props:      {
+    value: {
+      type:     Array,
+      default: () => {
+        return [];
+      }
+    },
+
+    names: {
+      type:    Array,
+      default: () => {
+        return [];
+      }
+    },
+
+    type: {
+      type:     String,
+      required: true
+    }
+  },
+
+  data() {
+    return {
+      errors:      [],
+      forceDelete: true
+    };
+  },
+
+  computed: {
+    ...mapState('action-menu', ['toRemove']),
+    ...mapGetters({ t: 'i18n/t' }),
+
+    plusMore() {
+      const count = this.names.length - this.names.length;
+
+      return this.t('promptRemove.andOthers', { count });
+    },
+
+    podNames() {
+      return this.names.reduce((res, name, i) => {
+        if (i >= 5) {
+          return res;
+        }
+        res += `<b>${ name }</b>`;
+        if (i === this.names.length - 1) {
+          res += this.plusMore;
+        } else {
+          res += i === this.toRemove.length - 2 ? ' and ' : ', ';
+        }
+
+        return res;
+      }, '');
+    },
+  },
+
+  methods: {
+    async forceRemovePod(resource) {
+      const opt = {
+        url:    resource.linkFor('self'),
+        method: 'delete',
+        data:   {
+          gracePeriod: 0,
+          force:       true
+        }
+      };
+
+      const res = await resource.$dispatch('request', { opt, type: resource.type } );
+
+      if ( res?._status === 204 ) {
+        // If there's no body, assume the resource was immediately deleted
+        // and drop it from the store as if a remove event happened.
+        return this.$dispatch('ws.resource.remove', { data: resource });
+      }
+
+      return res;
+    },
+
+    async remove(confirm) {
+      const parentComponent = this.$parent.$parent.$parent;
+
+      let goTo;
+
+      if (parentComponent.doneLocation) {
+        // doneLocation will recompute to undefined when delete request completes
+        goTo = { ...parentComponent.doneLocation };
+      }
+
+      try {
+        await Promise.all(this.value.map(resource => this.removePod(resource)));
+        if ( goTo && !isEmpty(goTo) ) {
+          parentComponent.currentRouter.push(goTo);
+        }
+        parentComponent.close();
+      } catch (err) {
+        parentComponent.error = err;
+        confirm(false);
+      }
+    },
+
+    removePod(pod) {
+      if (this.forceDelete) {
+        return this.forceRemovePod(pod);
```

Comment:
I don't think `forceRemovePod` is needed. You can pass through `data` in `opt` directly to the `remove` function. So something like 

```
      const opts = this.forceDelete ? {
        gracePeriod: 0,
        force:       true
      } : undefined;

      return pod.remove(opts);
```